pkg/resources/procedure.go

```diff
@@ -0,0 +1,464 @@
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/pkg/errors"
+)
+
+var procedureSchema = map[string]*schema.Schema{
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Specifies the identifier for the procedure; does not have to be unique for the schema in which the procedure is created. Don't use the | character.",
+	},
+	"database": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The database in which to create the procedure. Don't use the | character.",
+		ForceNew:    true,
+	},
+	"schema": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The schema in which to create the procedure. Don't use the | character.",
+		ForceNew:    true,
+	},
+	"arguments": {
+		Type: schema.TypeList,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:     schema.TypeString,
+					Required: true,
+					// Suppress the diff shown if the values are equal when both compared in lower case.
+					DiffSuppressFunc: DiffTypes,
+					Description:      "The argument name",
+				},
+				"type": {
+					Type:     schema.TypeString,
+					Required: true,
+					// Suppress the diff shown if the values are equal when both compared in lower case.
+					DiffSuppressFunc: DiffTypes,
+					Description:      "The argument type",
+				},
+			},
+		},
+		Optional:    true,
+		Description: "List of the arguments for the procedure",
+		ForceNew:    true,
+	},
+	"return_type": {
+		Type:        schema.TypeString,
+		Description: "The return type of the procedure",
+		// Suppress the diff shown if the values are equal when both compared in lower case.
+		DiffSuppressFunc: DiffTypes,
+		Required:         true,
+		ForceNew:         true,
+	},
+	"execute_as": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "OWNER",
+		Description: "Sets execute context - see caller's rights and owner's rights",
+	},
+	"null_input_behavior": {
+		Type:     schema.TypeString,
+		Optional: true,
+		Default:  "CALLED ON NULL INPUT",
+		ForceNew: true,
+		// We do not use STRICT, because Snowflake then in the Read phase returns RETURNS NULL ON NULL INPUT
+		ValidateFunc: validation.StringInSlice([]string{"CALLED ON NULL INPUT", "RETURNS NULL ON NULL INPUT"}, false),
+		Description:  "Specifies the behavior of the procedure when called with null inputs.",
+	},
+	"return_behavior": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		Default:      "VOLATILE",
+		Required:     false,
+		ForceNew:     true,
+		ValidateFunc: validation.StringInSlice([]string{"VOLATILE", "IMMUTABLE"}, false),
+		Description:  "Specifies the behavior of the function when returning results",
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "user-defined function",
```

Comment:
from the docs, it looks like this should be `stored procedure`

Reply:
No, the default comment for a stored procedure is "user-defined function". Its wierd but I tested it on Snowflake many times.